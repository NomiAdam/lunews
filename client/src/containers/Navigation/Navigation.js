import React, { lazy, memo, useContext } from 'react';
import PropTypes from 'prop-types';
import { Container, lightTheme, FadeIn } from 'bachelor-ui';
import { UiContext } from '../../context/uiContext';
import SuspenseLoader from '../../components/SuspenseLoader';

const GroupList = lazy(() => import('../GroupList'));
const Setting = lazy(() => import( '../Setting'));

const containerStyle = { borderRight: `1px solid ${ lightTheme.lightGrey }` };
const MemoizedContainer = memo(({ isMobile, setting }) => (
    <Container style={ isMobile ? undefined : containerStyle } height="100vh">
        <SuspenseLoader>
            <FadeIn>
                { setting ? <Setting /> : <GroupList isMobile={ isMobile } /> }
            </FadeIn>
        </SuspenseLoader>
    </Container>
), (prevProps, nextProps) => prevProps.isMobile === nextProps.isMobile && prevProps.setting === nextProps.setting);

const Navigation = ({ isMobile }) => {
    const { state: { setting } } = useContext(UiContext);
    return (
        <MemoizedContainer isMobile={ isMobile } setting={ setting } />
    );
};

Navigation.propTypes = {
    isMobile: PropTypes.bool,
};

export default Navigation;
