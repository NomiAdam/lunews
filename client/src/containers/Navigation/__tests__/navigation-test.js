/* eslint-disable no-undef,import/no-named-as-default */
import React from 'react';
import { shallow } from 'enzyme';
import { Navigation } from '../Navigation';

const mockFavourite = () => new Promise(resolve => resolve);
const mockList = () => new Promise(resolve => resolve);

describe('Navigation', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });
    it('should be defined', () => {
        expect(Navigation).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<Navigation listFavourite={ mockFavourite } listGroups={ mockList } />)).toMatchSnapshot();
    });
    it('should have default state defined', () => {
        expect(shallow(<Navigation listFavourite={ mockFavourite } listGroups={ mockList } />).state()).toEqual({
            groups: [],
            favouriteGroups: [],
        });
    });
});
