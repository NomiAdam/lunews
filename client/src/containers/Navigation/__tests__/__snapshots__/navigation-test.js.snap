// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match the snapshot 1`] = `
<styled.div
  gridGap="10px"
  padding="10px"
>
  <styled.div
    colXS={12}
  >
    <NavigationProfile
      favouriteCount={0}
      groupCount={0}
    />
  </styled.div>
  <styled.div
    colXS={12}
  >
    <Divider
      color="#bbbbbb"
      type="dashed"
    />
  </styled.div>
  <Component
    favouriteGroups={Array []}
    groups={Array []}
    handleGroupChange={[Function]}
  />
</styled.div>
`;
